### PR TITLE
TST: fix some mpl warnings

### PR DESCRIPTION
**Pull request by mzeitlin11**
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

This change fixes `UserWarning` being emitted in tests calling `autocorrelation_plot`, `andrews_curves`, and `radviz`. 

`UserWarning` was caused by calling `plt.gca(xlim=..., ylim=...)` on a potentially already existing axis, so `matplotlib` warns about this forcing new axis creation. Solution was just to explicitly use `set_xlim` and `set_ylim` instead.

Does it make sense to also add to the relevant tests an assertion that no UserWarning is emitted to guarantee the new behavior?

### CODE REVIEW

File: pandas/plotting/_matplotlib/misc.py [hunk 1]
```diff
@@ -144,7 +144,9 @@ def normalize(series):
     df = frame.drop(class_column, axis=1).apply(normalize)
 
     if ax is None:
-        ax = plt.gca(xlim=[-1, 1], ylim=[-1, 1])
+        ax = plt.gca()
+        ax.set_xlim([-1, 1])
+        ax.set_ylim([-1, 1])
```
Comment (ivanovmg) [inline pandas/plotting/_matplotlib/misc.py:149]: I guess you can even use ``ax.set_xlim(-1, 1)``, without extra brackets.
The same for other cases as well.

**Review by ivanovmg — COMMENTED** [review]
Please see the comment below.
I agree that addition of the tests is necessary.

**Review by jreback — CHANGES_REQUESTED** [review]
yeah pls add a tests that asssert_produces_warning(None) and fails on master but passes here

**Comment by mzeitlin11** [comment]
Will add tests once #38620 is figured out

File: pandas/plotting/_matplotlib/misc.py (hunk 1 above)
Comment (mzeitlin11) [reply pandas/plotting/_matplotlib/misc.py]: Done

**Comment by jreback** [comment]
> Will add tests once #38620 is figured out

hmm `tm.assert_produces_warning(None)`  should just work

**Comment by mzeitlin11** [comment]
> > Will add tests once #38620 is figured out
> 
> hmm `tm.assert_produces_warning(None)` should just work

It wasn't raising even when a warning was returned, with examples of this given in the issue. So there are currently no failures in master when wrapping these plotting calls with `tm.assert_produces_warning(None)`, even though a warning is returned

**Comment by mzeitlin11** [comment]
Wrapped existing calls that give warnings on master to avoid adding more plotting calls. If splitting out new tests for this is preferred instead, please let me know

**Comment by jreback** [comment]
> Wrapped existing calls that give warnings on master to avoid adding more plotting calls. If splitting out new tests for this is preferred instead, please let me know

this looks good; ping on green

**Review by jreback — APPROVED** [review]

**Comment by mzeitlin11** [comment]
Green

**Comment by jreback** [comment]
thanks @mzeitlin11

**Pull request merged**